src/middlewared/middlewared/api/v25_04_0/kerberos.py

```diff
@@ -0,0 +1,156 @@
+from middlewared.api.base import (
+    BaseModel,
+    NonEmptyString,
+    single_argument_args,
+)
+from middlewared.utils.directoryservices.krb5_constants import (
+    krb5ccache,
+)
+from pydantic import Field, Secret
+from typing import Literal
+
+
+__all__ = [
+    'KerberosKdestroyArgs', 'KerberosKdestroyResult',
+    'KerberosKinitArgs', 'KerberosKinitResult',
+    'KerberosKlistArgs', 'KerberosKlistResult',
+    'KerberosCheckTicketArgs', 'KerberosCheckTicketResult',
+    'KerberosGetCredArgs', 'KerberosGetCredResult',
+]
+
+
+class KerberosCredentialUsernamePassword(BaseModel):
+    """ Private API entry defined for normalization purposes """
+    username: NonEmptyString
+    password: Secret[NonEmptyString]
+
+
+class KerberosCredentialKeytab(BaseModel):
+    """ Private API entry defined for normalization purposes """
+    kerberos_principal: NonEmptyString
+
+
+class KerberosCcacheOptions(BaseModel):
+    """ Private API entry defined for normalization purposes """
+    ccache: Literal[
```

Comment:
We could consider `ccache: krb5ccache = krb5ccache.SYSTEM` as long as we handle the enum fields properly in the endpoints.